drivers/include/cst816s.h

```diff
@@ -0,0 +1,152 @@
+ */
+
+/**
+ * @defgroup    drivers_cst816s Cst816S touch screen driver
+ *
+ * @ingroup     drivers_sensors
+ * @brief       Device driver interface for the Hynitron CST816S touch screen
+ *
+ * The CST816S is a touch sensor from Hynitron with integrated gesture
+ * detection. It is able to measure both the position of a single finger and a
+ * number of basic gestures. The PineTime board has one of these for the touch
+ * screen.
+ *
+ * Documentation about the specifics is very limited and most of this driver is
+ * based on experimenting with the chip and from community effort on the
+ * PineTime.
+ *
+ * Two things about the driver are noteworthy:
+ *  1. It only responds to I2C commands after an event, such as a touch
+ *     detection. Do not expect it to respond on init. Instead after a touch
+ *     event, it will assert the IRQ and respond to I2C reads for a short time.
+ *  2. While it should be able to detect multiple finger events, this version of
+ *     the chip always returns only a single finger event and a gesture.
+ *
+ * Reading the display data multiple times during a single event will return the
+ * last sampled finger position.
+ *
+ * @{
+ * @file
+ * @brief       Device driver interface for the CST816S touch screen
+ *
+ * @author      koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef CST816S_H
+#define CST816S_H
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief cst816s_t forward declaration
+ */
+typedef struct _cst816s cst816s_t;
+
+/**
+ * @brief callback definition
+ */
+typedef void (*cst816s_irq_cb_t)(cst816s_t *dev, void *arg);
+
+/**
+ * @brief cst816s touch event touch state
+ */
+typedef enum {
+    CST816S_TOUCH_DOWN      = 0,    /**< Touch press */
+    CST816S_TOUCH_UP        = 1,    /**< Touch release */
+    CST816S_TOUCH_CONTACT   = 2,    /**< Touch contact */
+} cst816s_touch_t;
+
+/**
+ * @brief CST816S Gesture types
+ */
+typedef enum {
+    CST816S_GESTURE_NONE            = 0x00, /**< no gesture detecte        */
+    CST816S_GESTURE_SLIDE_DOWN      = 0x01, /**< downward slide detected   */
+    CST816S_GESTURE_SLIDE_UP        = 0x02, /**< upward slide detected     */
+    CST816S_GESTURE_SLIDE_LEFT      = 0x03, /**< left slide detected       */
+    CST816S_GESTURE_SLIDE_RIGHT     = 0x04, /**< right slide detected      */
+    CST816S_GESTURE_SINGLE_CLICK    = 0x05, /**< single click detected     */
+    CST816S_GESTURE_DOUBLE_CLICK    = 0x0b, /**< double click detected     */
+    CST816S_GESTURE_LONG_PRESS      = 0x0c, /**< long press detected       */
+} cst816s_gesture_t;
+
+/**
+ * @brief cst816s touch event data
+ */
+typedef struct {
+    cst816s_gesture_t gesture;  /**< Detected gesture */
+    cst816s_touch_t action;     /**< Press or release event */
+    uint16_t x;                 /**< X coordinate */
+    uint16_t y;                 /**< Y coordinate */
+} cst816s_touch_data_t;
+
+typedef struct {
```

Comment:
There is a missing `@brief` for that struct

Reply:
Added